modules/thermalmanager.c

```diff
@@ -52,545 +156,995 @@
 {
-  /* first send an indication to D-Bus */
-  {
-      DsmeDbusMessage* sig =
-          dsme_dbus_signal_new(path,
-                               interface,
-                               thermalmanager_state_change_ind);
-      dsme_dbus_message_append_string(sig, current_status_name());
-      dsme_dbus_signal_emit(sig);
-      dsme_log(LOG_NOTICE, "thermalmanager: Device (%s) thermal status: %s (%dC)", sensor_name, current_status_name(), temperature);
-  }
-
-  /* then broadcast an indication internally */
-  {
-      static bool temp_warning_sent = false;
-
-      if (current_status == THERMAL_STATUS_FATAL) {
-          send_thermal_status(DSM_THERMAL_STATUS_OVERHEATED, sensor_name, temperature);
-          temp_warning_sent = true;
-          dsme_log(LOG_CRIT, "thermalmanager: Device (%s) overheated (%dC)", sensor_name, temperature);
-      } else if (current_status == THERMAL_STATUS_LOW) {
-          send_thermal_status(DSM_THERMAL_STATUS_LOWTEMP, sensor_name, temperature);
-          temp_warning_sent = true;
-          dsme_log(LOG_WARNING, "thermalmanager: Device (%s) temperature low (%dC)", sensor_name, temperature);
-      } else if (temp_warning_sent) {
-          send_thermal_status(DSM_THERMAL_STATUS_NORMAL, sensor_name, temperature);
-          temp_warning_sent = false;
-          dsme_log(LOG_NOTICE, "thermalmanager: Device (%s) temperature back to normal (%dC)", sensor_name, temperature);
-      }
-  }
+    /* Ignore invalid / unregistered objects
+     */
+    if( !thermal_manager_object_is_registered(thermal_object) )
+        goto EXIT;
+
+    /* Schedule the next measurement point
+     */
+    DSM_MSGTYPE_WAIT msg = DSME_MSG_INIT(DSM_MSGTYPE_WAIT);
+
+    msg.req.pid = 0;
+    msg.data    = thermal_object;
+
+    /* Start with fall back defaults */
+    int mintime = THERMAL_STATUS_POLL_DELAY_DEFAULT_MINIMUM;
+    int maxtime = THERMAL_STATUS_POLL_DELAY_DEFAULT_MAXIMUM;
+
+    if( thermal_object_status_in_transition(thermal_object) ) {
+        /* In transition: multiple measurements are neede to
+         * verify the status change - wake up more frequently */
+        mintime = THERMAL_STATUS_POLL_DELAY_TRANSITION_MINIMUM;
+        maxtime = THERMAL_STATUS_POLL_DELAY_TRANSITION_MAXIMUM;
+
+        /* and wake up from suspend to do the measurement */
+        msg.req.wakeup  = true;
+    }
+    else if( !thermal_object_get_poll_delay(thermal_object,
+                                            &mintime, &maxtime) ) {
+        /* No wait period defined in the configuration - use
+         * shorter waits for abnormal states */
+
+        THERMAL_STATUS    status = THERMAL_STATUS_INVALID;
+        int               temperature = INVALID_TEMPERATURE;
+
+        thermal_object_get_sensor_status(thermal_object, &status,
+                                         &temperature);
+
+        switch( status ) {
+        case THERMAL_STATUS_ALERT:
+        case THERMAL_STATUS_FATAL:
+            mintime =   5;
```

Comment:
`5` and `10` look like good candidates for named constants at the top of the file (and include the unit, e.g. `_S` for seconds or `_MS` for milliseconds, etc.. in the name).